### DISCUSSION TITLE
How to get the separate quad(quad_like,rect_like) in PDF using search_for method when targeted letters are contiguous and adjacent to each other

### DISCUSSION THREAD

**Discussion by Carnegie-eng** [pymupdf/PyMuPDF#2806] [Q&A] [answered]
Hi, I would like to know how to get the separate quad(quad_like,rect_like) in PDF using search_for method when targeted letters are contiguous and adjacent to each other.

Default method of search_for(target) retrunes just 1 quad(quad_like,rect_like) since target letters are contiguous and adjacent to each other. I would like to get the separate quad(quad_like,rect_like).

**Comment by Carnegie-eng** [comment]
I wonder the quad area by search_for method is changed to be narrow in horizontal.  If so, could you let me know how to set the narrow. Thank you.

**Comment by JorjMcKie** [comment]
Like the text extraction variants, the search method also supports the `clip` parameter: `page.search_for(needle, clip=rect_like)`.
Using this in combination with `page.get_text("dict", clip=...)` or "rawdict" may help you bygoing down even to single characters.

In addition, there is the new parameter `delimiters` in `page.get_text("words", clip=rect, delimiters="§$%...")`, which allows to specify additional characters to be used as word delimiters: where the normal word extraction returns one single word `john.doe@outlook.com`, setting `delimiters="@."` will deliver 4 words "john", "doe", "outlook", "com" - together with their respective coordinates.

Maybe this helps.

**Comment by Carnegie-eng** [comment]
[test_copy.pdf](https://github.com/pymupdf/PyMuPDF/files/13349114/test_copy.pdf)

Thank you for advise. I tried like below to the attached example pdf. Target character is "9".

rect = page.search_for("9", clip=rect_like)
(Result) rect is only 8 area although there is 12 of "9" in the page.

I appreciate if you could advise me how to resolve this issue.

**Reply by JorjMcKie** [reply, marked as answer]
1-char searches are always problematic of course, because characters in a row may have tiny overlaps (created by the PDF maker). What you always do in similar cases is this:
```python
fitz.TOOLS.set_small_glyph_heights(True)
True
rl=page.search_for("9")
for r in rl:
    page.get_textbox(r)

    
'99'
'99'
'r\n9'
'9'
'9'
'9'
'99'
'99'
```
The second hickup: if the search algorithm finds multiple adjacent copies of the needle, then one common rectangle is returned as visible above.

**Comment by Carnegie-eng** [comment]
Thank you for explanation. I need just 1-char, so I will try another aproach. Thank you.
